src/libs/actions/Policy.ts

```diff
@@ -737,22 +805,28 @@ function clearAvatarErrors(policyID) {
 /**
  * Optimistically update the general settings. Set the general settings as pending until the response succeeds.
  * If the response fails set a general error message. Clear the error message when updating.
- *
- * @param {String} policyID
- * @param {String} name
- * @param {String} currency
  */
-function updateGeneralSettings(policyID, name, currency) {
-    const policy = allPolicies[`${ONYXKEYS.COLLECTION.POLICY}${policyID}`];
-    const distanceUnit = _.find(_.values(policy.customUnits), (unit) => unit.name === CONST.CUSTOM_UNITS.NAME_DISTANCE);
-    const distanceRate = _.find(_.values(distanceUnit ? distanceUnit.rates : {}), (rate) => rate.name === CONST.CUSTOM_UNITS.DEFAULT_RATE);
-    const optimisticData = [
+function updateGeneralSettings(policyID: string, name: string, currency: string) {
+    const policy = allPolicies?.[`${ONYXKEYS.COLLECTION.POLICY}${policyID}`];
+
+    if (!policy) {
+        return;
+    }
+
+    const distanceUnit = Object.values(policy?.customUnits ?? {}).find((unit) => unit.name === CONST.CUSTOM_UNITS.NAME_DISTANCE);
+    const distanceRate = Object.values(distanceUnit?.rates ?? {}).find((rate) => rate.name === CONST.CUSTOM_UNITS.DEFAULT_RATE);
+
+    if (!distanceUnit?.customUnitID || !distanceRate?.customUnitRateID) {
```

Comment:
Are you sure that if there's no `distanceUnit` or `distanceRate` the function should stop execution? As I can see from [the code below](https://github.com/Expensify/App/blob/46e4bba34c5837a493171fabe8af8e36abff829d/src/libs/actions/Policy.ts#L841) the data supports the case when there is no `distanceUnit`.

Reply:
I reviewed the function and you are correct, I have moved this check into the specific `optimisticData` and `failureData` assignments though, since they do require those to work.